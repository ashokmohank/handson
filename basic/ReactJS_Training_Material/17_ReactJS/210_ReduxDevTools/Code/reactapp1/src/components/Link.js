import React from 'react';

/*
This component will get two properties:
active: Whether link is active or not.
onLinkClick: Callback method to call when link is clicked.
*/

function Link(props) {
  let linkElement;

  function clickHandler(e) {
    e.preventDefault();
    props.onLinkClick();
  }

  if (props.active) {
    linkElement = <span>{props.children}</span>;
  } else {
    linkElement = (
      <a href="#" onClick={clickHandler}>
        {props.children}
      </a>
    );
  }

  return linkElement;
}

export default Link;
