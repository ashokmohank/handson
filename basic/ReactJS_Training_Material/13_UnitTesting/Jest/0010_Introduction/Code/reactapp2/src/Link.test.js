import React from "react";
import Link from "./Link";
import renderer from "react-test-renderer";

describe("Testing <Link/>", () => {
  const component = renderer.create(
    <Link page="http://www.facebook.com">Facebook</Link>
  );

  test("Component loads", () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Style changes on onMouseEnter", () => {
    // manually trigger the callback
    let tree = component.toJSON();
    tree.props.onMouseEnter();

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Style reverts to original on onMouseLeave", () => {
    // manually trigger the callback
    let tree = component.toJSON();
    tree.props.onMouseLeave();

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
