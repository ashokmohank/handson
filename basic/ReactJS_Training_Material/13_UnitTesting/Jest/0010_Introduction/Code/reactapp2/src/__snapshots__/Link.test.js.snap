// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Link/> Component loads 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Testing <Link/> Style changes on onMouseEnter 1`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Testing <Link/> Style reverts to original on onMouseLeave 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
